src/prefect/utilities/proxy.py

```diff
@@ -0,0 +1,58 @@
+import os
+from typing import (
+    Any,
+    Generator,
+)
+from urllib.parse import urlparse
+
+from python_socks.async_.asyncio import Proxy
+from typing_extensions import Self
+from websockets.client import WebSocketClientProtocol
+from websockets.legacy.client import Connect
+
+
+class WebsocketProxyConnect(Connect):
+    def __init__(self: Self, uri: str, **kwargs: Any):
+        # super() is intentionally deferred to the __proxy_connect__ method
+        # to allow for the socket to be established first
+
+        self.uri = uri
+        self.__kwargs = kwargs
```

Comment:
We don't have this double underscore convention anywhere else, so a single underscore should suffice.
```suggestion
        self._kwargs = kwargs
```

Reply:
I thought I needed this to avoid conflicts with the parent, but I'm realizing this basically doesn't matter because the `super()` is happening last once all this is done anyways